pandas/tests/test_expressions.py

```diff
@@ -332,78 +331,81 @@ def testit():
+        with pytest.raises(NotImplementedError, match=err_msg):
+            f(df.a, df.b)
 
-            with pytest.raises(NotImplementedError, match=err_msg):
-                f(df.a, True)
+        with pytest.raises(NotImplementedError, match=err_msg):
+            f(df.a, True)
 
-            with pytest.raises(NotImplementedError, match=err_msg):
-                f(False, df.a)
+        with pytest.raises(NotImplementedError, match=err_msg):
+            f(False, df.a)
 
-            with pytest.raises(NotImplementedError, match=err_msg):
-                f(False, df)
+        with pytest.raises(NotImplementedError, match=err_msg):
+            f(False, df)
 
-            with pytest.raises(NotImplementedError, match=err_msg):
-                f(df, True)
+        with pytest.raises(NotImplementedError, match=err_msg):
+            f(df, True)
 
-    def test_bool_ops_warn_on_arithmetic(self):
+    @pytest.mark.parametrize(
+        "op,name", list(zip(["+", "*", "-"], ["add", "mul", "sub"]))
+    )
+    def test_bool_ops_warn_on_arithmetic(self, op, name):
         n = 10
         df = DataFrame({"a": np.random.rand(n) > 0.5, "b": np.random.rand(n) > 0.5})
-        names = "add", "mul", "sub"
-        ops = "+", "*", "-"
+
         subs = {"+": "|", "*": "&", "-": "^"}
         sub_funcs = {"|": "or_", "&": "and_", "^": "xor"}
-        for op, name in zip(ops, names):
-            f = getattr(operator, name)
-            fe = getattr(operator, sub_funcs[subs[op]])
-
-            if op == "-":
-                # raises TypeError
-                continue
-
-            with tm.use_numexpr(True, min_elements=5):
-                with tm.assert_produces_warning(check_stacklevel=False):
-                    r = f(df, df)
-                    e = fe(df, df)
-                    tm.assert_frame_equal(r, e)
-
-                with tm.assert_produces_warning(check_stacklevel=False):
-                    r = f(df.a, df.b)
-                    e = fe(df.a, df.b)
-                    tm.assert_series_equal(r, e)
-
-                with tm.assert_produces_warning(check_stacklevel=False):
-                    r = f(df.a, True)
-                    e = fe(df.a, True)
-                    tm.assert_series_equal(r, e)
-
-                with tm.assert_produces_warning(check_stacklevel=False):
-                    r = f(False, df.a)
-                    e = fe(False, df.a)
-                    tm.assert_series_equal(r, e)
-
-                with tm.assert_produces_warning(check_stacklevel=False):
-                    r = f(False, df)
-                    e = fe(False, df)
-                    tm.assert_frame_equal(r, e)
-
-                with tm.assert_produces_warning(check_stacklevel=False):
-                    r = f(df, True)
-                    e = fe(df, True)
-                    tm.assert_frame_equal(r, e)
+
+        f = getattr(operator, name)
+        fe = getattr(operator, sub_funcs[subs[op]])
+
+        if op == "-":
```

Comment:
Does this even need to be parametrized if we just return?

Reply:
I'm planning on re-visiting this in an upcoming pass to see if it can be made viable.